### PR TITLE
[12.0][IMP] interface fixes

### PR DESCRIPTION
**Pull request by foutoucour**
Several fixes in the interface.

* organisations
before:
numigi - Odoo (20191111_1611)

after:
numigi - Odoo (20191111_1807)

* odoo modules
before:
project_stage - Odoo (20191111_1704)

after:
Odoo (20191111_1711)

It also fixes the smart buttons and the related IDs that are passed to the filtering (ID instead of active_id)

### CODE REVIEW

**Comment by foutoucour** [comment]
@legalsylvain another PR for the beast

**Comment by pedrobaeza** [comment]
I see them correctly on Odoo community.

**Comment by foutoucour** [comment]
hello @pedrobaeza 
You are right. The current update will make them correctly rendered also in enterprise, so in all Odoo versions ;)

File: github_connector/views/view_github_organization.xml [hunk 1]
```diff
@@ -45,33 +45,35 @@ License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
...
-                            <field name="member_qty" string="Members" widget="statinfo" />
+                        <button type="action" name="%(action_github_repository)d"
+                                class="oe_stat_button" icon="fa-folder-o"
+                                attrs="{'invisible': [('repository_qty', '=', False)]}"
+                                context="{'search_default_organization_id': active_id}">
+                            <field name="repository_qty" string="Repositories" widget="statinfo" />
                         </button>
                     </div>
+
+                    <field name="image" widget="image" class="oe_avatar"/>
```
Comment (pedrobaeza) [inline github_connector/views/view_github_organization.xml:70]: Why are you changing the side of the image?

File: github_connector/views/view_github_repository.xml [hunk 2]
```diff
@@ -45,13 +45,13 @@ License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
...
                                 attrs="{'invisible': [('repository_branch_qty', '=', False)]}"
-                                context="{'search_default_repository_id': active_id}">
+                                context="{'search_default_repository_id': id}">
                             <field name="repository_branch_qty" string="Branches" widget="statinfo" />
                         </button>
                         <button type="action" class="oe_stat_button" icon="fa-users"
                                 name="%(action_github_team_repository_from_repository)d"
                                 attrs="{'invisible': [('team_qty', '=', False)]}"
-                                context="{'search_default_repository_id': active_id}">
+                                context="{'search_default_repository_id': id}">
```
Comment (legalsylvain) [inline github_connector/views/view_github_repository.xml:54]: Id is working since wich version ? I allways saw only active_id.

**Review by legalsylvain — APPROVED** [review]
One non blocking question. Lgtm otherwise.

File: github_connector/views/view_github_repository.xml (hunk 2 above)
Comment (foutoucour) [reply github_connector/views/view_github_repository.xml:54]: good question. I gave it a try and worked ;)

File: github_connector/views/view_github_organization.xml (hunk 1 above)
Comment (foutoucour) [reply github_connector/views/view_github_organization.xml:70]: All other views had the avatar at the right side. I did it to standardize the form views. 
What do you think ?

Comment (pedrobaeza) [reply github_connector/views/view_github_organization.xml:70]: Standard Odoo put it images on the left if I'm not wrong. Can you confirm? If so, I prefer to standardize the Odoo way.

File: github_connector/views/view_github_repository.xml (hunk 2 above)
Comment (pedrobaeza) [reply github_connector/views/view_github_repository.xml:54]: #28 is related to this when you navigate for example from organization to repository and then to branch, taking incorrect `active_id`. I'm not sure if `id` is the solution to this. If not, we should change this to methods that put proper context by code using `self`.

File: github_connector/views/view_github_organization.xml (hunk 1 above)
Comment (foutoucour) [reply github_connector/views/view_github_organization.xml:70]: Actually, we were confused by my screenshots of entreprise interface.
With the previous code, the avatar was forced to be on the left side on CE and EE. With the new version of the code, the avatar is placed by the CSS.
Here are the screenshot on CE version with the new code:

Account Budget Template - Odoo (20191112_0939)
account_budget_oca - Odoo (20191112_0939)
numigi - Odoo (20191112_0939)

It looks standard on CE and EE now. Both world is happy.
What do you think?

Comment (pedrobaeza) [reply github_connector/views/view_github_organization.xml:70]: Great then, thanks for checking!

Offtopic: do you really prefer enterprise interface over `web_responsive` OCA module? We have included a lot of extra features that are not on enterprise one.

Comment (foutoucour) [reply github_connector/views/view_github_organization.xml:70]: @pedrobaeza not my choice :) Client picked EE ;)

We did some test with the new CE interface and its enhanced and it gave us some arguments to have some of our new clients on CE.
But this on is on EE, for now ;)

Comment (pedrobaeza) [reply github_connector/views/view_github_organization.xml:70]: Well, IMO it doesn't matter: we have enterprise customers too, but without that module installed, and installing `web_responsive` instead.

File: github_connector/views/view_github_repository.xml (hunk 2 above)
Comment (foutoucour) [reply github_connector/views/view_github_repository.xml:54]: @pedrobaeza @legalsylvain 
unfortunately the trick doesn't work on #28 issue.
I won't have the time to work on it yet, and I would prefer to not over-extend the scope of the PR.

Would you agree ?

Comment (pedrobaeza) [reply github_connector/views/view_github_repository.xml:54]: OK, then please keep `active_id` for now

Comment (foutoucour) [reply github_connector/views/view_github_repository.xml:54]: @pedrobaeza I understand your concern, but I would not like to revert the change.
With `active_id` the behaviour of the instance is not correct while it is with the `id`. 
I can't see any issue with the update. Could you let me know what they are ? Then I will see if I can fix them

@legalsylvain do you see any blocking issues with `id` ?

Comment (pedrobaeza) [reply github_connector/views/view_github_repository.xml:54]: But I don't get what is the problem with `active_id`. Do you get some steps to reproduce the problem?

Comment (foutoucour) [reply github_connector/views/view_github_repository.xml:54]: Hello @pedrobaeza 
in our case `active_id` have the wrong behaviour.

Here is the behaviour the PR fixes:
From an organisation, go the repository using the smart button
odooOrga-repos

Then you go to the repository branches, again using the smart button
account-budgeting - Odoo (20191120_1000)

From here, you will see that the filter is based on another repository.
Repository Branches - Odoo (20191120_1000)

It is due to `active_id` referring to `github_repository.id` and to `github_repository_branch.id`. To have the proper behaviour, in the case of a deep digging using smart button, it seems it is better to use `id`. 
It is a wild case I confess.

Does it make sense ?

Comment (pedrobaeza) [reply github_connector/views/view_github_repository.xml:54]: Yes, but this is the same problem that it's depicted in the referred issue, and the proper solution is to switch buttons to be method based. If you are not able to do it, please remove this and I'll put a PR.

**Comment by legalsylvain** [comment]
@foutoucour could you revert regarding id / active_id ? 

this module is ported by @tecnativa team to V13, and should be great to keep that commits.

**Comment by pedrobaeza** [comment]
FYI I have asked my colleague to fix the active_id stuff in v13

**Comment by pedrobaeza** [comment]
The smart-buttons navigation has been fixed in v13 through methods, so I close this as incorrect approach.

**Pull request closed without merging**